src/oss-defog/DefoggerTool.cs

```diff
@@ -159,8 +160,8 @@ public async Task AnalyzePackage(PackageURL purl)
         {
             Logger.Trace("AnalyzePackage({0})", purl.ToString());
 
-            var downloadTool = new DownloadTool();
-            foreach (var directory in await downloadTool.Download(purl))
+            var packageDownloader = new PackageDownloader(purl, (string)(this.Options["cache-directory"]));
```

Comment:
Actually why does AnalyzePackage have two constructors?

Wouldn't `AnalyzePackage(PacakageURL purl, string? directory = null)` and then pass through directory to `PackageDownloader` work just as well?

That would also get rid of this case where you implicitly check the cache directory.

Reply:
I guess they are separate functions because the tool might need the ability to analyze a directory that is not a package - not sure what is the right change here

Reply:
Pushed a commit that fixes this so the constructor matches the others with the directory and cache options.